datasync/chaindatafetcher/api.go

```diff
@@ -47,11 +47,11 @@ func (api *PublicChainDataFetcherAPI) Status() string {
 }
 
 func (api *PublicChainDataFetcherAPI) ReadCheckpoint() (int64, error) {
-	return api.f.repo.ReadCheckpoint()
+	return api.f.checkpointDB.ReadCheckpoint()
 }
 
 func (api *PublicChainDataFetcherAPI) WriteCheckpoint(checkpoint int64) error {
```

Comment:
Consider this next. :)
```suggestion
func (api *PublicChainDataFetcherAPI) WriteCheckpoint(checkpoint uint64) error {
```

Reply:
Okay. I will make a ticket for this.